editor/src/components/inspector/resize-to-fit-control.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { FlexRow, Icn, Tooltip } from '../../uuiui'
+import { EditorAction, EditorDispatch } from '../editor/action-types'
+import { useDispatch } from '../editor/store/dispatch-context'
+import { Substores, useEditorState } from '../editor/store/store-hook'
+import { setPropHugStrategies } from './inspector-strategies/inspector-strategies'
+import { executeFirstApplicableStrategy } from './inspector-strategies/inspector-strategy'
+
+export const ResizeToFitControlTestId = 'ResizeToFitControlTestId'
+
+interface ResizeToFitControlProps {}
+
+export const ResizeToFitControl = React.memo<ResizeToFitControlProps>(() => {
+  const dispatch = useDispatch()
+  const selectedViews = useEditorState(
```

Comment:
you can make it a never-rerendering component if you make the selectedViews and metadata refEditorStates! since you only need it in the callback anyways.

Reply:
Thanks for spotting it! Updated to use refs in https://github.com/concrete-utopia/utopia/pull/3154/commits/7aa95563d3493e736fdeaf1b1859da9a6b7bf59e